pkg/test/framework/components/echo/docker/instance.go

```diff
@@ -162,7 +162,8 @@ func (i *instance) WaitUntilCallable(instances ...echo.Instance) error {
 		}
 	}
 
-	if !i.cfg.Annotations.GetBool(echo.SidecarInject) {
+	// Note: docker version environment implementation only supports single container.
+	if !i.cfg.Workloads[0].Annotations.GetBool(echo.SidecarInject) {
```

Comment:
Do you need to loop through all the workloads to see if any have this unset? Can we have only some workloads with this annotation?

Reply:
see my comments above. Even before the PR, i tried the test I used. Docker env seems failing multiple containers deployment for some reason. I can add a TODO for investigation there.

Reply:
ah ok.  Yeah file a bug and reference it in a todo

Reply:
done.